src/canopy/knowledge_base/record_encoder/hybrid.py

```diff
@@ -0,0 +1,111 @@
+from typing import List
+from functools import cached_property
+from pinecone_text.dense.base_dense_ecoder import BaseDenseEncoder
+from pinecone_text.hybrid import hybrid_convex_scale
+from pinecone_text.sparse import BM25Encoder
+from .base import RecordEncoder
+from canopy.knowledge_base.models import KBQuery, KBEncodedDocChunk, KBDocChunk
+from canopy.models.data_models import Query
+
+
+class HybridRecordEncoder(RecordEncoder):
+    """
+    HybridRecordEncoder is a subclass of RecordEncoder that generates sparse and dense vector representation of
+    documents` chunks and textual queries.
+
+    The dense representation generated by the `HybridRecordEncoder` is a list of floats in a given dimension.
+    The sparse representation generated by the `HybridRecordEncoder` is a `SparseVector`.
+
+    HybridRecordEncoder uses BaseDenseEncoder for dense encoding and BM25Encoder for sparse encoding.
+
+    Alpha is a parameter that controls the weight of the dense vector in the hybrid representation.
+    If alpha is 1, the query vector will be the dense vector, if alpha is 0, the query vector will be the sparse vector.
+    The default value of alpha is 0.5.
+
+    For more information about the encoders see: https://github.com/pinecone-io/pinecone-text
+
+    """  # noqa: E501
+
+    def __init__(self,
+                 dense_encoder: BaseDenseEncoder,
+                 alpha: float = 0.5,
+                 **kwargs):
+        """
+        Initialize the encoder.
+
+        Args:
+            dense_encoder: A BaseDenseEncoder to encode the text.
+            alpha: The weight of the dense vector in the hybrid representation (between 0 and 1).
+            **kwargs: Additional arguments to pass to the RecordEncoder.
+        """  # noqa: E501
+
+        if not 0 <= alpha <= 1:
+            raise ValueError("Alpha must be between 0 and 1")
+
+        super().__init__(**kwargs)
+        self._dense_encoder = dense_encoder
+        self._sparse_encoder = BM25Encoder.default()
```

Comment:
IMO we should support loading a user file from day 1 because this is what we recommend also to users. It simply adding an optional file path and call `load` if it's not none

Reply:
Done